go-controller/pkg/kube/annotator.go

```diff
@@ -32,6 +32,8 @@ type nodeAnnotator struct {
 }
 
 // NewNodeAnnotator returns a new annotator for Node objects
+// Node annotator compares current annotations found in the node(*kapi.Node) and only updates them if there is a change.
+// It does not sync the node(*kapi.Node) with the API server which means that in most cases it should be Run() once and discarded.
```

Comment:
OK but that means we need to follow that everywhere, i.e don't pass around a `NodeAnnotator` And instead just create them on command... with this commit we start splitting how we use a NodeAnnotator object I think we should keep it consistent around the code..... Why not just make the NodeAnnotator go fetch the most up to date version of the node Anytime it's run? SO that is only "syncs on command".  Then we can continue to just pass it around like we do everywhere else.

Reply:
> OK but that means we need to follow that everywhere, i.e don't pass around a NodeAnnotator And instead just create them on command... with this commit we start splitting how we use a NodeAnnotator object I think we should keep it consistent around the code..... 

AFAIK in all other cases we are passing the `NodeAnnotator` only to call `Set` on it and it is run once.

>  Why not just make the NodeAnnotator go fetch the most up to date version of the node Anytime it's run? SO that is only "syncs on command". Then we can continue to just pass it around like we do everywhere else.

This was my initial approach but after a quick chat with @danwinship I went with what is in the PR. I am ok to change that and swap `node(*kapi.Node)` wind `nodeName(string)` in `NewNodeAnnotator`.